### DISCUSSION TITLE
HA Core su Pi4 e 2021.09.* --> Problemi

### DISCUSSION THREAD

**Discussion by riccionese** [indomus/forum#681] [HUB: Home Assistant]
Ho letto il post https://github.com/indomus/forum/discussions/673 la discussione cui rimanda ma io continuo ad avere problemi.

Come sempre installo le nuove rel quando escono e così ho fatto anche per la 2021.9.0 ma HA non ripartiva per cui ho reinstallato la 2021.8.4 e tutto è tornato a funzionare

Ho riprovato con la 2021.9.1 ma ho lo stesso problema e quindi ... di nuovo 2021.8.4 e tutto a posto

suggerimenti ?

**Comment by fizzi** [comment]
Come prima cosa ti consiglio di guardare i log, cosi stai andando alla cieca.
Guarda i log e cerca di capire l'errore.

**Reply by riccionese** [reply]
Riprovato.
un ` $ tail -f /home/homeassistant/.homeassistant/home-assistant.log` non si muove di una riga 
il  `sudo systemctl status home-assistant@homeassistant`  dice questo, con le prime tre righe (PID 16700, 16733 e 16760) fisse mentre le 

![image](https://user-images.githubusercontent.com/62329292/132002029-9e2b49ec-29b7-440c-9101-60e5b1e1f758.png)

altre si ripetono a nastro 
sono arrivato a vedere anche 4 `gcc`

per cui torno alla 2021.8.4
<niubbo_mode> oppure ci sono altri log dovrei guardare ? </niubbo_mode>

**Reply by riccionese** [reply]
Ovviamente sto riferendomi alla 2021.9.1 ... che avevo capito risolvesse "qualcosa"

**Reply by snakuzzo** [reply]
Se sei su Core non dovresti avere i problemi legati a quella issue.
Prima di fare qualsiasi cosa e prima di un restart ti consiglio di rinominare temporaneamente la directory `custom_components`, in modo tale che al successivo restart non li caricherà e potrai escludere eventuali problemi relativi appunto ai custom component

Poi...se vuoi vedere i log relativi al servizio, apri una shell e dai...

`sudo journalctl -f -u home-assistant@homeassistant.service`

e nel frattempo su altra shell riavvia HA.
A questo punto vedi sulla prima shell cosa logga

Altra cosa che puoi fare è mettere i log di HA a debug con [logger](https://www.home-assistant.io/integrations/logger/)

```yaml
logger:
  default: debug
```

**Reply by riccionese** [reply]
Io odio queste cose !!! 
ho seguito il tuo consiglio, come al solito.
ho rinominato la `custom_component` ed ho riavviato HA ---> E' PARTITO !!!
ovviamente con entità invalide ma è partito.

ho quindi rimesso al suo posto la `custom_component` con l'idea di far ripartire ed andare in errore per poi, una alla volta, rimuovere le directory contenute fino a trovare quella che "spacca" HA.

Ebbene ... rimessa al suo posto la `custom_cpmponent` senza toccare un bit e riavviato HA ... è ripartito come niente fosse, tutto perfetto
Io odio quando a parità di condizioni le cose cambiano.
Oppure odio quando non capisco che condizioni sono cambiate tra il "prima" ed il "dopo".

Ma come al solito, grazie Diego!

**Reply by snakuzzo** [reply]
Figurati...contento che tu abbia risolto.

Certo è un bel mistero. L'unica cosa che mi viene da pensare è che forse era memorizzato qualcosa sotto `.storage` e che. rimuovendo prima e rimettendo poi la directory, quel qualcosa sia cambiato.
Senza log a debug è difficile capire cosa sia potuto accadere.

Se ricapita ti conviene impostare prima il log a debug e vedere cosa accade

**Reply by riccionese** [reply]
Cercherò di tenerlo a mente.
In ogni caso, mistero per mistero, anche il fatto che la 2021.8.4 funziona nelle stesse condizioni in cui la 2021.9.1 no ... non è male!
E non può dipendere dal fatto che installando la 2021.9.1 si rompa qualcosa nei dati/configurazioni, perché tornando alla 2021.8.4, questa ricomincia a funzionare
mah !

![image](https://user-images.githubusercontent.com/62329292/132027007-2f907bdb-9d0c-41c3-a713-a1ff0e91c909.png)

Ciaux !

**Comment by pitaccop** [comment]
Scusate la mancanza di esperienza: io ho Hassio che sta usando la versione core 2021.9.0. Da interfaccia web non riesco a fare nulla perché appena torna la connessione la stessa viene persa e non riesco ad impartire nessun comando.
Mi sono collegato fisicamente con monitor e tastiera al Raspberry PI3 per eseguire l'update alla 2021.9.1 sperando risolva o per fare un downgrade, ma al comando "core update" o "core update --version 2021.9.1" o "core update --version 2021.8.8" mi da come risultato del processo "DONE" però la versione rimane la stessa e ovviamente il problema persiste, anche dopo il restart. 
Suggerimenti? 
Grazie in anticipo

**Reply by snakuzzo** [reply]
Devi stare più attento alle "news" e alle evoluzioni...
Leggi [QUI](https://github.com/home-assistant/core/issues/55541)

Con la 2021.9.0 c'è il problema che si riavvia di continuo.

**Reply by pitaccop** [reply]
Grazie mille, snakuzzo. 
Si avevo letto il post che mi hai linkato e anche qualcun altro (infatti poi ho sbagliato anche discussione mettendola in quella del PI4). 
In realtà essendo un neofita non sapevo bene come seguire le indicazioni da terminale... sarei stato (forse) in grado di seguire i passaggi da interfaccia web, ma non riuscivo ad usarla a causa del loop di riavvii continui.
Comunque, a forza di tentare, alla fine il CORE si è aggiornato alla versione 2021.9.1 (è da ieri dopo cena che stavo provando!!!) e ora tutto ha ripreso a funzionare.

Grazie ancora per il pronto supporto e buona serata.
P.

**Reply by snakuzzo** [reply]
di nulla ;)

**Comment by 50494554524F** [comment]
ormai sono entrati nella fase di rilasci compulsivi senza debuggare niente... non aggiornerò più subito... tenetevi una iso da parte nel caso togliessero le vecchie

ps: ieri era impossibile usare la ui nonostante i riavvii (2021.7.x), evidentemente hanno anche qualcosa che richiama qualche loro server è blocca tutto... da indagare

**Reply by snakuzzo** [reply]
Non è che fanno rilasci compulsivi, si seguono le ormai definite date di rilascio, che prevedono una major release al mese. Ovvio che poi ci siano varie fix e ben vengano pure 10 rilasci di fila se servono a risolvere cose che non vanno.
I problemi grossi (se così vogliamo chiamarli) ci sono stati sulla 2021.9.0 e 2021.9.1 ma solo in determinate circostanze...e non sulla 2021.7.x che risale a luglio.
La cosa del "c'è qualcosa che richiama il loro server" da dove esce fuori? Se hai qualcosa che non va imposti il log a debug e vedi cosa accade.
Parliamo di un software che, nonostante l'impegno di chi sviluppa per fare in modo che qualsiasi utente sia in grado di usarlo facendo due click, va seguito costantemente, studiato e capito...ed alcune nozioni di base a mio avviso bisogna averle.
Inutile commentare arrivando a conculsioni affrettate senza sapere di cosa si sta parlando.
Poi ovvio che poi ognuno è libero di fare ciò che vuole e, per chi non sa bene come muoversi in questi casi, sicuramente è una buona cosa avere un backup pronto per ogni evenienza.

**Reply by 50494554524F** [reply]
- non è la prima volta che il raspberry va in loop, sarà la terza da quando uso homeassistant. se non controlli il raspberry che è sicuramente usato dal 100% dei tuoi utenti non pratici vuol dire rilasci compulsivi.
- se il frontend non è usabile per tutta una giornata nonostante vari riavvii vuol dire che richiama qualcosa che poi lo blocca. il frontend si auto aggiorna senza intervento, quindi già contatta i loro server... devo solo verificare con uno sniffer se lo fa durante tutte le 24h bloccandosi se c'è downtime dei loro server. come avrete notato, quando esce una nuova versione, presumibilmente a causa degli aggiornamenti degli utenti, il frontend è lento anche se gira in locale...
- sento spesso questa cosa delle nozioni base, va bene... ma deve essere usabile anche da chi non vuole, sennò il meccanico ripara la macchina solo a chi conosce la fisica termodinamica e via.... ora immagina chi non è pratico e deve reinstallare tutto da zero ogni volta per non andare a riga di comando...
- certo che è gratis ma allora non rilasciare stabile se non hai controllato niente, inspiegabile questo mtodo...
- spero di avere spiegato sapendo di cosa sto parlando...

**Reply by snakuzzo** [reply]
Sì, hai spiegato il tuo punto di vista. Ci sta pure che ti stranisci se aggiorni e non funziona più niente. Allo stesso tempo però dovresti sapere come muoverti per tornare indietro ed uscire da queste situazioni.
HA è anche questo...un progetto in continua evoluzione con i suoi pro e i suoi contro.
Se vuoi un prodotto fatto e finito che ad ogni aggiornamento ad occhi chiusi ti assicuri il funzionamento al 100%, hai sbagliato software.
Prova con altri hub.

Come dice un mio amico "HA è per tutti, ma non per tutti"

**Reply by snakuzzo** [reply]
Solo per precisare che la mia non è una polemica. Ognuno ovviamente è libero di fare quello che vuole e pensare quello che vuole.
Ti ho solo descritto il mio punto di vista dopo circa 2 anni e mezzo di esperienza.

**Reply by KC-inDomus** [reply]
(l'amico sono io)

**Comment by Juri78** [comment]
Ciao a tutti.
Ho iniziato una installazione da zero di Home Assistant Core su Raspberry Pi4.
Seguendo la guida ho installato il tutto e funziona, ma la versione di HA Core che mi installa in automatico è la 2021.1.5 non va oltre, nel blog mi compare il messaggio:
2021-09-22 05:58:34 WARNING (MainThread) [homeassistant.bootstrap] Support for the running Python version 3.7.3 is deprecated and will be removed in the first release after December 7, 2020. Please upgrade Python to 3.8.0 or higher.

Ho aggiornato Python all'ultima versione disponibile, quindi la 3.9.7 ma HA mi compare sempre lo stesso messaggio; inoltre eseguendo i comandi di aggiornamento di HA non mi installa l'ultima versione.

Come posso fare per aggiornare all'ultima revisione?
Grazie a chi mi dedicherà un pò di tempo.

**Reply by snakuzzo** [reply]
Perché hai installato nel venv vecchio.
Quando aggiorni python devi ricreare il venv e reinstallare homeassistant nel nuovo venv, passandogli poi la config che già avevi (quindi non perdi nulla)...cerca sul sito...abbiamo spiegato come fare

**Reply by Juri78** [reply]
Grazie mille, risolto... finalmente ho HA Core all'ultima versione disponibile.

Ti chiedo un'ultima cosa: mi devo preoccupare di questi due errori che mi sono risultati durante l'upgrade di HA?

ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.
We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.
requests 2.25.1 requires idna<3,>=2.5, but you'll have idna 3.2 which is incompatible.

WARNING: You are using pip version 20.2.3; however, version 21.2.4 is available.
You should consider upgrading via the '/srv/homeassistant/bin/python3.8 -m pip install --upgrade pip' command.

Grazie ancora.
